src/wallet/wallet.cpp

```diff
@@ -2573,9 +2575,20 @@ bool CWallet::CreateTransaction(const vector<CRecipient>& vecSend, CWalletTx& wt
 
     wtxNew.fTimeReceivedIsTxTime = true;
     wtxNew.BindWallet(this);
+    int nextBlockHeight = chainActive.Height() + 1;
     CMutableTransaction txNew = CreateNewContextualCMutableTransaction(
-        Params().GetConsensus(), chainActive.Height() + 1);
+        Params().GetConsensus(), nextBlockHeight);
 
+    // Activates after Overwinter network upgrade
+    // Set nExpiryHeight to 20 blocks (default) past current blockheight
+    if (NetworkUpgradeActive(nextBlockHeight, Params().GetConsensus(), Consensus::UPGRADE_OVERWINTER)) {
+        if (nextBlockHeight + blockExpiry > TX_EXPIRY_HEIGHT_THRESHOLD){
```

Comment:
`if (nextBlockHeight + blockExpiry >= TX_EXPIRY_HEIGHT_THRESHOLD) {`

Reply:
Non-blocking: file a ticket for abstracting this into a common atomic check (i.e. either the provided output variable is set, or the function returns false), so we aren't relying on this check being correct everywhere.